### DISCUSSION TITLE
Lambda function XXX failed to stabilize since it is in InProgress state

### DISCUSSION THREAD

**Discussion by matmeylan** [aws/aws-cdk#14053] [General]
We've been experiencing issues with `DockerImageFunction`. All deployments fail with the following (and very cryptic to me) error:

> Lambda function XXX failed to stabilize since it is in InProgress state

Other functions deploy without any problems. Deployments were working fine until Wednesday 07.04.2021. Since then, they fail every time. We haven't changed anything in our CDK code, in that function Dockerfile or its code.

We deploy using cdk typescript. I tested with 1.93 and 1.97 (latest version at the time of this writing).

Any clue ?

**Comment by bigkraig** [comment]
Same thing, opened a support req with AWS and they are currently at a loss. There is also a stack overflow where some people are describing this problem at https://stackoverflow.com/questions/67008190/lambda-function-failed-to-stabilize-since-it-is-in-inprogress-state.

**Reply by wjossey** [reply]
@mmeylan I work with @bigkraig and we'll update here as soon as we hear anything from them. Getting back on a call with them shortly.

**Comment by wjossey** [comment]
@mmeylan This issue should be resolved. You'll need to force complete the rollback, and then redeploy.

**Reply by matmeylan** [reply]
Great ! Any detail on the root cause ?

**Comment by cmckni3** [comment]
Ran into the same problem today on CDK `1.114.0`.


> Resource handler returned message: "Resource of type 'AWS::Lambda::Function' with identifier 'XXX' did not stabilize." (RequestToken: XXX, HandlerErrorCode: NotStabilized)
